### ISSUE TITLE
S3 presign signature validation fails and tags are missing for multipart uploads

### ISSUE THREAD

**Issue by efenderbosch** [localstack/localstack#3713]
# Type of request:

This is a bug report.

# Detailed description
* S3 multipart presign signature validation fails using latest Docker image (as of 2021-03-12), works using 0.12.7
* multipart uploads do not have tags after upload has been completed

## Expected behavior
* S3 multipart presign signature validation should work the same as S3 (and previous versions of localstack)
* completed multipart uploads should have tags present

## Actual behavior
* S3 multipart presign signature validation fails using latest Docker image (as of 2021-03-12)
* multipart uploads do not have tags after upload has been completed

# Steps to reproduce
Minimally reproducible project here: https://github.com/efenderbosch/localstack-multipart-upload-test
Follow steps in readme.md.

## Command used to start LocalStack
docker-compose -f docker-compose.yml up
```yaml
version: '3'

services:
  localstack:
    image: localstack/localstack-light:latest
    # image: localstack/localstack-light:0.12.7
    ports:
      - '4566:4566'
    environment:
      SERVICES: s3
      DEBUG: 1

networks:
  default:
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
Using AWS Java SDK 2.15 and 2.16 in Kotlin 1.4 project. See project above for full code.

**Comment by whummer** [comment]
Thanks for reporting @efenderbosch . This should (hopefully) be fixed in #3760 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!

**Comment by efenderbosch** [comment]
Now I'm getting a 500 error w/ multipart uploads when sending a part using the pre-signed URL.

 Traceback:
```
localstack_1  | 2021-03-24T21:10:41:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request PUT http://localhost:4566/iq-files-test/638f7f85-8ce5-11eb-ae37-115fdd2ebb1f/6390b806-8ce5-11eb-ae37-37da7840aa7a/6392dae7-8ce5-11eb-ae37-e304deffd2c7/test/random.bin?partNumber=1&uploadId=8amFl7L8NeDZsxks7Nia6hnTLcuK4sRPdnFwNK4uyqTZCiaZCAVjOWQ&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210324T211041Z&X-Amz-SignedHeaders=host&X-Amz-Expires=900&X-Amz-Credential=test%2F20210324%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=5b586e653d8d730a7112bef7f7000d228e6fcfd352e7d24704b10fe4dff992ae: 'utf-8' codec can't decode byte 0xdb in position 0: invalid continuation byte Traceback (most recent call last):
localstack_1  |   File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 129, in index
localstack_1  |     raise result
localstack_1  |   File "/opt/code/localstack/localstack/utils/bootstrap.py", line 671, in run
localstack_1  |     result = self.func(self.params)
localstack_1  |   File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
localstack_1  |     return fn(*args, **kwargs)
localstack_1  |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 607, in handler
localstack_1  |     response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
localstack_1  |   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 373, in modify_and_forward
localstack_1  |     listener_result = listener.forward_request(method=method,
localstack_1  |   File "/opt/code/localstack/localstack/services/edge.py", line 56, in forward_request
localstack_1  |     auth_header = get_auth_string(method, path, headers, data)
localstack_1  |   File "/opt/code/localstack/localstack/services/edge.py", line 181, in get_auth_string
localstack_1  |     data_components = parse_request_data(method, path, data)
localstack_1  |   File "/opt/code/localstack/localstack/utils/common.py", line 833, in parse_request_data
localstack_1  |     result = parse_qs(to_str(data or ''))
localstack_1  |   File "/opt/code/localstack/localstack/utils/common.py", line 984, in to_str
localstack_1  |     return obj.decode(encoding, errors) if isinstance(obj, six.binary_type) else obj
localstack_1  | UnicodeDecodeError: 'utf-8' codec can't decode byte 0xdb in position 0: invalid continuation byte
```

**Comment by leikoilja** [comment]
Having the same issue :/

**Comment by ysholomii** [comment]
I am facing the same issue.

**Comment by whummer** [comment]
Thanks for the update @efenderbosch @leikoilja @ysholomii . Can you please verify if the issue still exists with the `latest` Docker image? If you can provide a small, isolated example we'll be happy to look into it in more detail. Thanks!

**Comment by leikoilja** [comment]
Looks like `latest` works perfectly now, thanks a lot, @whummer

**Comment by efenderbosch** [comment]
Latest works for signing, but multipart tags still do not work.

**Comment by whummer** [comment]
Thanks for the update @efenderbosch . The multipart tagging issue should now be fixed in #3951 (available in the `latest` Docker image). Please let us know if the problem persists. Thanks!
